package/MDAnalysis/lib/util.py

```diff
@@ -2165,6 +2166,33 @@ def wrapper(*args, **kwargs):
         return wrapper
     return check_coords_decorator
 
+def check_atomgroup_not_empty(groupmethod):
+    """Decorator triggering a ``ValueError`` if the underlying group is empty.
+
+    Avoids obscure computational errors on group methods.
+
+    Raises
+    ------
+    ValueError
+        If the :class:`~MDAnalysis.core.groups.AtomGroup`,
```

Comment:
Does this work on all groups? If so we should change the name and test. Otherwise we need to change docstring.